api/src/main/java/eu/darkbot/api/game/entities/FakeEntity.java

```diff
@@ -0,0 +1,58 @@
+package eu.darkbot.api.game.entities;
+
+import eu.darkbot.api.game.other.Location;
+
+/**
+ * Fake entities help to keep virtual entities on a map that placed out of visibility range.
+ * It should use when need to assist or predict position these entities for active module.
+ * Cases of use: keeping mine locations for choose a better path and prevent releasing it,
+ *               keeping box locations that should be collected after more prefer tasks.
+ */
+public interface FakeEntity extends Entity {
+    static boolean isFakeEntity(Entity entity) {
+        return entity instanceof FakeEntity;
+    }
+
+    /**
+     * Set timeout lifetime of a fake object in ms, -1 - disable timer
```

Comment:
```suggestion
     * Set timeout lifetime of a fake object in ms
```
I'd avoid the option of no-bound timed fake entities, let's always assume they *will* go away unless the timeout is constantly refreshed (this means that if someone was to disable the plugin feature, even if no explicit cleanup happens, they would *eventually* go away)

Reply:
Feature with infinite timer works good with mines, the mine removing when ship located nearly mine, in my case it works correctly while agatus event.